src/hotspot/share/code/codeBlob.cpp

```diff
@@ -315,6 +316,56 @@ AdapterBlob* AdapterBlob::create(CodeBuffer* cb) {
   return blob;
 }
 
+MethodHandleIntrinsicBlob::MethodHandleIntrinsicBlob(
+  Method* method,
+  int mhi_size,
+  CodeBuffer* code_buffer):
+  RuntimeBlob("MethodHandle intrinsic", code_buffer, sizeof(MethodHandleIntrinsicBlob), mhi_size, CodeOffsets::frame_never_safe, 0, NULL),
+  _method(method) {
+  {
+    debug_only(NoSafepointVerifier nsv;)
+    assert_locked_or_safepoint(CodeCache_lock);
+
+    assert(code_buffer->total_oop_size() == 0, "unexpected oop");
+    assert(code_buffer->total_metadata_size() == 0, "unexpected metadata");
+
+    CodeCache::commit(this);
+  }
+}
+
+void* MethodHandleIntrinsicBlob::operator new(size_t s, int size) throw() {
+  return CodeCache::allocate(size, CodeBlobType::NonNMethod);
+}
+
+MethodHandleIntrinsicBlob* MethodHandleIntrinsicBlob::create(const methodHandle& method,
+  CodeBuffer *code_buffer) {
+  code_buffer->finalize_oop_references(method);
+
+  MethodHandleIntrinsicBlob* mhi = NULL;
+  {
+    MutexLocker mu(CodeCache_lock, Mutex::_no_safepoint_check_flag);
+    int mhi_size = CodeBlob::allocation_size(code_buffer, sizeof(MethodHandleIntrinsicBlob));
+
+    mhi = new (mhi_size) MethodHandleIntrinsicBlob(method(), mhi_size, code_buffer);
+  }
+
+  if (mhi != NULL) {
+    debug_only(mhi->verify();) // might block
+  }
```

Comment:
This is debug only. Looking at CodeCache::allocate, it can only return `NULL` if the allocation size is `<= 0`, in which case an earlier assert will already fire. So, this null check doesn't seem needed?
```suggestion
  debug_only(mhi->verify();) // might block
```